mesheryctl/internal/cli/root/system/model.go

```diff
@@ -246,24 +246,119 @@ mesheryctl system model view [model-name]
 	},
 }
 
-// ModelCmd represents the `mesheryctl system model` command
+// represents the `mesheryctl exp model search [query-text]` subcommand.
+var searchModelCmd = &cobra.Command{
+	Use:   "search",
+	Short: "search models",
+	Long:  "search a models by search string",
+	Example: `
+// View current provider
+mesheryctl exp model search [query-text]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp model search [query-text]\nRun 'mesheryctl exp model search --help' to see detailed help message"
+		if len(args) == 0 {
+			return fmt.Errorf("query text must not be empty\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
```

Comment:
Are we able to use a MeshKit error here?

Either way, please enhance this error to be more palatable to users. What action should they take? This error should be standard across each command. Will you confirm? Let's make it consistent.

Reply:
@leecalcote `GetMesheryCtl` function [returns](https://github.com/meshery/meshery/blob/0dd426525ab7dda847dd02b5c0b1bbf4721c6127/mesheryctl/internal/cli/root/config/config.go#L70) meshkit error.